### PR TITLE
Update MarkCommand in DG

### PR DESCRIPTION
**Pull request by comicalromance**
- Update implementation details of `mark`.
- Include UML diagrams.
- Include design considerations.

### CODE REVIEW

File: docs/diagrams/MarkCommandSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+MarkCommand -> MarkCommand : executeNoRefresh()
+activate MarkCommand
+
+loop until end of indexes
+    MarkCommand -> StudentEditor : editStudent(studentToEdit)
+    activate StudentEditor
+
+    create StudentData
+    StudentEditor -> StudentData : studentToEdit.getStudentData()
```
Comment (highorbit25) [inline docs/diagrams/MarkCommandSequenceDiagram.puml:26]: are . notation legal for sequenceDiagram

File: docs/diagrams/MarkCommandSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    StudentData -> StudentEditor
+    deactivate StudentData
+
+    StudentEditor -> MarkCommand : editedStudent
+    deactivate StudentEditor
+
+    MarkCommand -> Model : setStudent(studentToEdit, editedStudent)
+    activate Model
+    Model -> MarkCommand
```
Comment (highorbit25) [inline docs/diagrams/MarkCommandSequenceDiagram.puml:41]: should return arrows be dotted instead?

**Review by highorbit25 — APPROVED** [review]
Overall LGTM as a draft DG, some minor formatting issues could be resolved in due time

**Comment by comicalromance** [comment]
Thanks for the comments bao bin! I'll fix them soon

**Pull request merged**